integration_tests/proxy/proxy.rb

```diff
@@ -1,27 +0,0 @@
-#! /usr/bin/env ruby
```

Comment:
I found it a lot easier to have this TCP server be inside the `rspec` process, so there isn't a process boundary to navigate. I think it would have been necessary for it to be a docker image before to get around weird networking sadness.